workers/unity/Packages/com.improbable.gdk.core/Worker/WorldCommandsReceivedStorage.cs

```diff
@@ -20,12 +21,23 @@ public class WorldCommandsReceivedStorage : ICommandDiffStorage
         private readonly MessageList<WorldCommands.EntityQuery.ReceivedResponse> entityQueryResponses =
             new MessageList<WorldCommands.EntityQuery.ReceivedResponse>();
 
+        private readonly Comparer comparer = new Comparer();
+
+        private bool createEntitySorted;
+        private bool deleteEntitySorted;
+        private bool reserveEntityIdsSorted;
+        private bool entityQueriesSorted;
+
         public void Clear()
         {
             createEntityResponses.Clear();
             deleteEntityResponses.Clear();
             reserveEntityIdsResponses.Clear();
             entityQueryResponses.Clear();
+            createEntitySorted = false;
```

Comment:
Can the `MessageList` not keep track of its sorted state?

Reply:
Yes it can. That is something that I'd like to add in. This was the minimal change to keep it consistent. What might happen is that we make these apply things in sorted order and that can turned off if you don't need it.